chain/chunks/src/logic.rs

```diff
@@ -0,0 +1,195 @@
+use std::sync::Arc;
+
+use near_chain::{validate::validate_chunk_proofs, Chain, ChainStore, RuntimeAdapter};
+use near_chunks_primitives::Error;
+use near_primitives::{
+    hash::CryptoHash,
+    merkle::{merklize, MerklePath},
+    receipt::Receipt,
+    sharding::{
+        EncodedShardChunk, PartialEncodedChunk, PartialEncodedChunkPart, PartialEncodedChunkV1,
+        PartialEncodedChunkV2, ReceiptProof, ShardChunk, ShardChunkHeader, ShardProof,
+    },
+    types::{AccountId, ShardId},
+};
+use tracing::log::{debug, error};
+
+#[derive(Clone)]
+pub struct ChunksLogic {
+    pub me: Option<AccountId>,
+    pub runtime_adapter: Arc<dyn RuntimeAdapter>,
+}
+
+impl ChunksLogic {
```

Comment:
I'm not sure if it simplifies the code by creating a new struct that includes AccountId and RuntimeAdapter. I would think just creating some static methods is enough and also easier to understand, especially that here the shared parameters are only two.